### PR TITLE
ci(circleci): added docker lint, source lint, unit test steps

### PR DESCRIPTION
**Pull request by xavidop**

### CODE REVIEW

**Review by frankgu968 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -134,12 +134,16 @@
...
     "local": "NODE_TLS_REJECT_UNAUTHORIZED=0 NODE_ENV=local ts-node --files -r tsconfig-paths/register start.ts",
     "local:watch": "NODE_TLS_REJECT_UNAUTHORIZED=0 NODE_ENV=local forever --watch -c \"ts-node --files -r tsconfig-paths/register\" start.ts",
     "preview:check": "echo \"generalruntime\"",
     "preview:new": "scripts/deploy_dev_environments.sh",
     "start": "NODE_ENV=production node --max-old-space-size=8192 build/start.js",
-    "test": "NODE_ENV=test nyc ts-mocha --paths --opts mocha.opts",
-    "test:single": "NODE_ENV=test ts-mocha --paths --opts mocha-single.opts"
+    "test": "yarn test:unit && yarn test:integration",
+    "test:integration": "NODE_ENV=test nyc ts-mocha --paths --opts ./config/tests/integration/mocha.it.opts",
+    "test:single": "NODE_ENV=test ts-mocha --paths --opts ./config/tests/single/mocha.single.opts",
```
Comment (frankgu968) [inline package.json:146]: Move the single test config to `./config/tests`

File: .circleci/config.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ defaults:
 
 orbs:
   codecov: codecov/codecov@1.0.2
-  vfcommon: voiceflow/common@0.0.25
+  vfcommon: voiceflow/common@0.0.43
```
Comment (frankgu968) [inline .circleci/config.yml:34]: Update to 0.0.44 or latest

**Review by frankgu968 — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (xavidop) [reply .circleci/config.yml]: done

File: package.json (hunk 1 above)
Comment (xavidop) [reply package.json]: done

**Review by leartgjoni-voiceflow — APPROVED** [review]

**Review by frankgu968 — APPROVED** [review]

**Pull request merged**